### ISSUE TITLE
hhsearch Segmentation Fault

### ISSUE THREAD

**Issue by ckallu** [soedinglab/hh-suite#249]
Hi, 

I am attempting to run hhsearch, however I am getting a segmentation error everytime. I installed via conda (conda install -c conda-forge -c bioconda hhsuite). My test case is a single fasta sequence. I downloaded the pfamA_33.1.tar.gz database from http://wwwuser.gwdg.de/~compbiol/data/hhsuite/databases/hhsuite_dbs/ and stored the unzipped files in ./database/. I even tried using -cpu 1 and -maxmem 1 (however, I checked and I my computing cluster has ample memory: -m 4194304 -v 4194304 -s unlimited -f unlimited) The command I run and the output I get is:

hhsearch -i ./pfam_inputs/single-fasta.faa -d ./database/pfam -i ./pfam_outputs/single-fasta.out

- 10:03:13.432 INFO: Search results will be written to ./pfam_inputs/XP_014772622.hhr

- 10:03:13.615 INFO: ./pfam_inputs/XP_014772622.1 is in A2M, A3M or FASTA format

- 10:03:13.618 INFO: Searching 18259 database HHMs without prefiltering

- 10:03:13.631 INFO: Iteration 1

- 10:03:14.047 INFO: Scoring 18259 HMMs using HMM-HMM Viterbi alignment

- 10:03:14.076 INFO: Alternative alignment: 0

- 10:03:26.977 INFO: 18259 alignments done

- 10:03:27.032 INFO: Alternative alignment: 1

- 10:03:27.143 INFO: 95 alignments done

- 10:03:27.143 INFO: Alternative alignment: 2

- 10:03:27.144 INFO: 1 alignments done

- 10:03:27.144 INFO: Alternative alignment: 3

Segmentation fault

**Comment by milot-mirdita** [comment]
Could you upload the `pfam_inputs/single-fasta.fa`  file? I can try to reproduce the issue locally.

**Comment by ckallu** [comment]
Actually I was able to resolve this by running it as a job on the cluster I am using as opposed to trying to run it on the login node. 

It was an issue with memory, I tried to run a fasta sequence of length 5 on the login node using the same command and it worked, but it gave a segmentation error when it was a 90AA sequence. I tried to change sysctl vm.overcommit_memory=1 but do not have the permissions to do so. But running it as a job using 5 compute nodes worked.
